pkg/resources/reconciling/ensure.go

```diff
@@ -60,6 +62,49 @@ func createWithName(rawcreate ObjectCreator, name string) ObjectCreator {
 	}
 }
 
+// EnsureObject is specification of the object to ensure.
+type EnsureObject struct {
+	Name             types.NamespacedName
+	Creator          ObjectCreator
+	EmptyObj         runtime.Object
+	RequiresRecreate bool
+}
+
+// EnsureNamedObjectsConcurrent calls `EnsureNamedObject` for each specifiecation concurrently and returns aggregated errors if any occurred.
+func EnsureNamedObjectsConcurrent(ctx context.Context, client ctrlruntimeclient.Client, objs []EnsureObject) error {
+	var wg sync.WaitGroup
+	errChan := make(chan error)
+	for _, obj := range objs {
+		wg.Add(1)
+		go func(obj EnsureObject) {
+			defer wg.Done()
+			err := EnsureNamedObject(ctx, obj.Name, obj.Creator, client, obj.EmptyObj, obj.RequiresRecreate)
+			if err != nil {
+				errChan <- fmt.Errorf("ensure %+v: %v", obj.Name, err)
+			}
+		}(obj)
+	}
+
+	ret := make(chan error)
+	defer close(ret)
```

Comment:
if we move `wg.Wait()` here, we woudn't need to launch a goroutine to gather errors.